Add DNS requirements (#26)
NethServer/ns8-docs: system_requirements.rst

.. _supported-distros-section:

Linux distribution
==================

Supported GNU/Linux distributions and versions:

- `Rocky Linux <https://rockylinux.org/>`_ 9
- `CentOS Stream <https://www.centos.org/centos-stream/>`_ 9
- `AlmaLinux <https://almalinux.org>`_ 9
- `Debian <https://www.debian.org/>`_ 12

.. _static-ip-reqs:

Static IP address
=================

> The server must be assigned a static IP address. DHCP and any other
> dynamic IP discovery protocols are not allowed.

Reviewer: ```suggestion
Assign a static IP address to the server. DHCP and any other
dynamic IP discovery protocols are not allowed.
```

.. _dns-reqs:

DNS configuration
=================

As you are configuring a server, network clients must resolve its fully
qualified domain name (FQDN) to a routable IP address with the DNS. This
is a requirement to connect with the server.

A correct FQDN and DNS setup is also a requirement for TLS encryption to
work properly. Once connected with the server, network clients check if

> the FQDN is in the TLS certificate before exchanging data with it.

Reviewer: ```suggestion
the TLS certificate is valid for the given FQDN.
```

1. Depending on your server purpose, DNS can be provided by a public
   internet service, a private network appliance, or even both of them.
   Read carefully and understand their documentation.

2. Decide the FQDN of your server and register it in the DNS with the
   server public IP address. A FQDN is composed by the host name prefix (a
   single word) and the DNS domain suffix. For instance, host name can be
   ``jupiter`` and domain suffix ``example.org``: the resulting FQDN is
   ``jupiter.example.org``.

.. _worker-node-reqs:

Worker node requirements
========================

.. highlight:: text

> Among network clients, a NethServer 8 worker node has some special

Reviewer: ```suggestion
Among network clients, a worker node has some special
```

requirements to be installed and configured.

> A worker node reaches the leader during the join procedure at the

Reviewer: ```suggestion
The worker node reaches the leader during the join procedure at the
```
